File: app/src/main/java/com/yechy/dailypic/ui/GalleryPagerAdaper.java
package com.yechy.dailypic.ui;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;

import com.github.chrisbanes.photoview.PhotoView;
import com.yechy.dailypic.R;
import com.yechy.dailypic.repository.PictureInfo;
import com.yechy.dailypic.util.ImageLoader;

import java.util.List;

/**
 * Created by cloud on 2019-12-20.
 */
public class GalleryPagerAdaper extends PagerAdapter {

    private Context mContext;
    private List<PictureInfo> pictureInfoList;

    public GalleryPagerAdaper(Context context, List<PictureInfo> pictureInfoList) {
        this.mContext = context;
        this.pictureInfoList = pictureInfoList;
    }

    @Override
    public int getCount() {
        if (pictureInfoList != null) {
            return pictureInfoList.size();
        }
        return 0;
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.layout_gallery_item, container, false);
        PhotoView photoView = view.findViewById(R.id.pv_gallery_item_image);
        if (pictureInfoList != null && pictureInfoList.size() > position) {
            PictureInfo pictureInfo = pictureInfoList.get(position);
            ImageLoader.load(mContext, photoView, pictureInfo.getUrl());
        }
        container.addView(view, ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);

        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((View) object);
    }
}

File: app/src/main/java/com/yechy/dailypic/ui/gallery/GalleryPagerAdaper.kt
package com.yechy.dailypic.ui.gallery

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.viewpager.widget.PagerAdapter
import com.github.chrisbanes.photoview.PhotoView
import com.yechy.dailypic.R
import com.yechy.dailypic.repository.PictureInfo
import com.yechy.dailypic.util.ImageLoader

/**
 * Created by cloud on 2019-12-20.
 */
class GalleryPagerAdaper(
    private val mContext: Context,
    private val pictureInfoList: MutableList<PictureInfo>
) : PagerAdapter() {

    fun setData(dataList: List<PictureInfo>) {
        pictureInfoList.clear()
        pictureInfoList.addAll(dataList)
    }

    override fun getCount(): Int {
        return pictureInfoList?.size ?: 0
    }

    override fun isViewFromObject(
        view: View,
        `object`: Any
    ): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view =
            LayoutInflater.from(mContext).inflate(R.layout.layout_gallery_item, container, false)
        val photoView: PhotoView = view.findViewById(R.id.pv_gallery_item_image)
        val textView = view.findViewById<TextView>(R.id.tv_gallery_item_copyright)
        if (pictureInfoList != null && pictureInfoList.size > position) {
            val (url, _, _, copyRight) = pictureInfoList[position]
            ImageLoader.load(mContext, photoView, url)
            textView.text = copyRight
        }
        container.addView(
            view,
            ViewGroup.LayoutParams.MATCH_PARENT,
            ViewGroup.LayoutParams.MATCH_PARENT
        )
        return view
    }

    override fun destroyItem(
        container: ViewGroup,
        position: Int,
        `object`: Any
    ) {
        container.removeView(`object` as View)
    }

}